lib/delayed/job_groups/job_group.rb

```diff
@@ -80,9 +94,26 @@ def ready_for_completion?
       end
 
       def complete
-        Delayed::Job.enqueue(on_completion_job, on_completion_job_options || {}) if on_completion_job
+        job = nil
+        job_options = nil
+
+        # Deserialization of the job or its options can fail
+        begin
+          job = on_completion_job
+          job_options = on_completion_job_options
+        rescue StandardError => e
+          Delayed::Worker.logger.info('Failed to deserialize the on_completion_job or on_completion_job_options for ' \
+                                      "job_group_id=#{id}. Skipping on_completion_job to clean up job group.")
+          error_reporter.call(e) if error_reporter
```

Comment:
I suppose an alternative here is that the job group could mark itself as blocked instead of destroying itself.

Reply:
Wouldn't we still need to manage some state to indicate why a job group was blocked e.g. so you could query the DB for the list of job groups that are blocked due to failure and unblock them after a fix has been deployed?

Reply:
It would certainly be more convenient if we did that. I think we could get away with using a single failed_at timestamp on the on_cancellation_job or the on_completion_job that would be set if deserialization failed. Blocking the job group still makes sense right?